R/checkPythonDeps.R

```diff
@@ -0,0 +1,350 @@
+#' checkPythonDeps.R
+#'
+#' Verify the availability of REMIND's Python dependencies on the host system. Exposes two functions:
+#' \code{checkPythonRequirements} and \code{checkPythonDeps}. While \code{checkPythonRequirements} reads a pip-style
+#' requirements file and compares the required packages with the installed ones, \code{checkPythonDeps} checks if the
+#' required Python dependencies can actually be imported in the Python environment provided.
+#'
+#' @author Tonn Rüter
+
+# ---------------------
+# Utility Functions
+# ---------------------
+
+#' Create Python Version
+#'
+#' This function creates a named list representing a Python version string from its components.
+#'
+#' @param major Major version number
+#' @param minor Minor version number
+#' @param patch Patch version number
+#' @param releaseType Release type (e.g. alpha, beta, rc)
+#' @param releaseVersion Release version number
+#' @return A named list representing a Python version string
+#' @importFrom reticulate import py_list_packages
+#' @importFrom stringr regex str_match
+#' @importFrom purrr map map_chr pmap
+#' @importFrom utils strcapture
+createPythonVersion <- function(major = "", minor = "", patch = "", releaseType = "", releaseVersion = "") {
+  # Store the components in a named list
+  return(list(
+    major = as.integer(major),
+    minor = as.integer(minor),
+    patch = as.integer(patch),
+    releaseType = as.character(releaseType),
+    releaseVersion = as.integer(releaseVersion)
+  ))
+}
+
+#' Print Python version string
+#'
+#' This function creates a string representation of a Python version.
+#'
+#' @param pythonVerion A named list representing the Python version.
+#' @return A string representation of the Python version.
+printPythonVersion <- function(pythonVerion) {
+  return(paste0(
+    ifelse(is.na(pythonVerion$major), "", as.character(pythonVerion$major)),
+    ifelse(is.na(pythonVerion$minor), "", paste0(".", as.character(pythonVerion$minor))),
+    ifelse(is.na(pythonVerion$patch), "", paste0(".", as.character(pythonVerion$patch))),
+    ifelse(is.na(pythonVerion$releaseType), "", pythonVerion$releaseType),
+    ifelse(is.na(pythonVerion$releaseVersion), "", pythonVerion$releaseVersion)
+  ))
+}
+
+#' Extract Python Version
+#'
+#' This function extracts a named list representing Python version from a string by matching a regular expression.
+#'
+#' @param versionString A Python version string like "3.10.0a0"
+#' @return A named list representing the Python version
+extractPythonVersion <- function(versionString) {
```

Comment:
Do we really have to do this ourselves? Maybe there's a python lib for that already that we could call via reticulate?

Reply:
c.f. https://github.com/pik-piam/piamenv/pull/19#discussion_r1731469447